projects/igniteui-angular/src/lib/time-picker/time-picker.directives.ts

```diff
@@ -257,6 +266,20 @@ export class IgxHourItemDirective {
         return this.isSelectedHour && this.itemList.isActive;
     }
 
+    @HostBinding('class.igx-time-picker__item--disabled')
+    get applyDisabledStyleForHours(): boolean {
+        if (this.minValueDate === undefined || this.maxValueDate === undefined) {
+            return false;
+        }
+        let hour = parseInt(this.value, 10);
+        if (this.selectedAmPm === 'PM') {
+            hour += 12;
+        }
+        const date = new Date(this.minValueDate);
+        date.setHours(hour);
+        return !(date <= this.maxValueDate && date >= this.minValueDate);
```

Comment:
Just for readability's sake.
```suggestion
        return date < this.minValueDate || this.maxValueDate < date;
```

#

Alternatevly, you could also use `greaterThanMinValue` and `lessThanMaxValue` methods from `DatePickerUtil`. They will spare you the need to check everything by hand:
```
return DatePickerUtil.lessThanMinValue(date, this.minValueDate)
    || DatePickerUtil.greaterThanMaxValue(date, this.maxValueDate);
```

Reply:
I went with the two methods from the DatePickerUtil even though they are a bit more verbose.